invokeai/app/services/model_install/model_install_default.py

```diff
@@ -287,6 +290,39 @@ def sync_to_config(self) -> None:
             self._logger.info(f"{len(installed)} new models registered")
         self._logger.info("Model installer (re)initialized")
 
+    def _migrate_yaml(self) -> None:
+        db_models = self.record_store.all_models()
+        try:
+            yaml = self._get_yaml()
+        except OSError:
+            return
+
+        if len(db_models) == 0 and len(yaml.items()) != 0:
+            self._logger.info(
+                f"Starting one-time migration of {len(yaml.items())} models in yaml file to DB. This may take a few minutes."
+            )
+
+            for model_key, stanza in yaml.items():
+                if model_key == "__metadata__":
+                    assert (
+                        stanza["version"] == "3.0.0"
+                    ), f"This script works on version 3.0.0 yaml files, but your configuration points to a {stanza['version']} version"
```

Comment:
should this be wrapped in a try/except? If they have a non-3.0.0 yaml file it should probably just not migrate

Reply:
Good idea, added